### ISSUE TITLE
smb folder listener

### ISSUE THREAD

**Issue by jajouka79** [icewind1991/SMB#53]
hi, does anyone know if its possible to set a listener on the smb folder that detects changes in the directory? thanks

**Comment by icewind1991** [comment]
Using version 2.x

```php
$share->notify('')->listen(function (\Icewind\SMB\Change $change) {
    echo $change->getCode() . ': ' . $change->getPath() . "\n";
});
```

Or using the older syntax with 1.1.x

```php
$share->notify('', function ($code, $path) {
    echo $code . ': ' . $path . "\n";
});
```

The meaning of the different `$code` values can be found [here](https://github.com/icewind1991/SMB/blob/master/src/INotifyHandler.php#L14), more information can be found in the phpdoc of the relevant methods.

**Comment by cowai** [comment]
@icewind1991 This worked for weeks, but now I get :
Uncaught TypeError: Argument 1 passed to {closure}() must be an instance of Icewind\SMB\Change, null given, called in vendor/icewind/smb/src/NotifyHandler.php on line 62 ...

Do you have any immediate solution for this error or do you need more details?

**Comment by icewind1991** [comment]
fixed with 2.0.1
